### PR TITLE
Support script[type=application/json] in templates

### LINKED ISSUE #12023: amp-mustache removes `application/json` script tags
Template sanitizer currently removes `<script type="application/json">` which means that components that rely on JSON configurations (`amp-animation`, `amp-ima-video`) can not be be used inside `amp-list`

/cc @dvoytenko  @choumx

### PR DESCRIPTION
**Pull request by choumx**
Fixes #12023.

### CODE REVIEW

**Comment by choumx** [comment]
I reran the [DOMPurify test suite](https://github.com/cure53/DOMPurify/blob/master/test/fixtures/expect.js) and there are no new breakages.

File: test/functional/test-purifier.js [hunk 1]
```diff
@@ -26,6 +26,36 @@ import {toggleExperiment} from '../../src/experiments';
...
+      const html = '<script type="application/javascript">alert(1)</script>';
+      expect(purifyHtml(html)).to.equal('');
+    });
+
+    it('should allow script[type="application/json"]', () => {
+      const html = '<script type="application/json">{}</script>';
+      expect(purifyHtml(html)).to.equal(html);
+      // Should not allow insecure <script> tags following a secure one.
+      expect(purifyHtml(html + '<script>alert(1)</script>')).to.equal(html);
+    });
```
[the remark this answers is not in the document]
Comment (jridgewell) [reply test/functional/test-purifier.js]: Nope, looking for explicitly allowed then disallowed, but adding another disallowed then allowed wouldn't hurt.

[the remark this answers is not in the document]
Comment (choumx) [reply test/functional/test-purifier.js]: Isn't that what the test already did? `expect(valid + invalid).to.equal(valid)`

**Pull request merged**